backend/modules/parsers/videoparser.py

```diff
@@ -0,0 +1,162 @@
+        clip = VideoFileClip(video_filepath)
+        temp_images_dir = tempfile.mkdtemp()
+        clip.write_images_sequence(
+            os.path.join(temp_images_dir, file_name + "_frame%04d.png"),
+            fps=0.2,  # configure this for controlling frame rate.
+        )
+        return temp_images_dir
+
+    def _get_audio_from_video(self, video_filepath: str) -> str:
+        """
+        Extract audio from video
+        """
+        _, file_name = os.path.split(video_filepath)
+        # Save the audio in a temporary directory
+        temp_audio_dir = tempfile.mkdtemp()
+        clip = VideoFileClip(video_filepath)
+        clip.audio.write_audiofile(os.path.join(temp_audio_dir, file_name + ".wav"))
+        return temp_audio_dir
+
+    async def get_chunks(
+        self, filepath: str, metadata: Dict[Any, Any] | None, **kwargs
+    ) -> List[Document]:
+        """
+        Get the chunks of the video file.
+        """
+        try:
+            # Extract images from video
+            temp_images_dir = self._get_images_from_video(filepath)
+
+            # Init multimodal parser for extracting text from images
+            multimodal_parser = MultiModalParser(
+                model_configuration=self.vlm_model_configuration, prompt=self.prompt
+            )
+
+            # Iterate over all the images and extract text from them
+            parsed_video_text = []
+            for idx, image in enumerate(os.listdir(temp_images_dir)):
+                image_path = os.path.join(temp_images_dir, image)
+                image_text = await multimodal_parser.get_chunks(image_path, metadata)
+                parsed_video_text.extend(image_text)
+                logger.info(f"Extracted text from image: {image_text[0].page_content}")
+                if idx == self.total_frames:
+                    break
+
+            # Extact audio from video
+            temp_audio_dir = self._get_audio_from_video(filepath)
+
+            # Init audio parser for extracting text from audio
+            audio_parser = AudioParser(
+                model_configuration=self.audio_model_configuration,
+                max_chunk_size=self.max_chunk_size,
+            )
+
+            # Extract text from audio
+            audio_text = await audio_parser.get_chunks(
+                os.path.join(temp_audio_dir, os.listdir(temp_audio_dir)[0]), metadata
+            )
+
+            # Combine all the extracted text
+            final_text = parsed_video_text + audio_text
+
+            # Remove temporary directories
+            try:
+                for temp_dir in [temp_images_dir, temp_audio_dir]:
+                    for file in os.listdir(temp_dir):
+                        os.remove(os.path.join(temp_dir, file))
+                    os.rmdir(temp_dir)
+                    logger.info(f"Removed temporary directory: {temp_dir}")
+            except Exception as e:
+                logger.exception(f"Error in removing temporary directories: {e}")
+
```

Comment:
Instead of manually managing these, use tempfile's context manager, to create a working directory that will be auto cleaned up before exiting get_chunks scope


```
async get_chunks(...):
    with tempfile.TemporaryDirectory(...) as directory:
        try:
            await self._get_chunks(...)
        ...

Reply:
Sure, would move to context manager.